### PR TITLE
Add two pingpong sequence modes for whitewhale.

### PR DESCRIPTION
**Pull request by GoneCaving**
Per this thread:
http://llllllll.co/t/white-whale-ping-pong-playback-mode/4816/41

mPing: 1,2,3,4,3,2,1,2...
mPingRep: 1,2,3,4,4,3,2,1,1,...

Seems to be working, pattern & series mode checked. I've not tested on a smaller grid, and i think there may be a clash with the keys from the left hand side? Happy to make any further changes to address this? (drop one mode?). It would help to know what the "FIXME" around line 882 relate to.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -807,14 +859,26 @@ static void handler_MonomeGridKey(s32 data) {
 					keyfirst_pos = x;
 				}
 				else if(key_alt == 1) {
-					if(x == LENGTH)
+					if(x == LENGTH) {
 						w.wp[pattern].step_mode = mForward;
-					else if(x == LENGTH-1)
+                    }
```
Comment (genericwoods) [inline src/main.c:864]: Minor: it looks like the alignment of these closing braces is kind of off.

File: src/main.c [hunk 2]
```diff
@@ -807,14 +859,26 @@ static void handler_MonomeGridKey(s32 data) {
...
-					else if(x == LENGTH-3)
+                    }
+					else if(x == LENGTH-3) {
 						w.wp[pattern].step_mode = mRandom;
+                    }
+					else if(x == LENGTH-4) {
+						w.wp[pattern].step_mode = mPing;
+                        w.wp[pattern].ping_dir = mPingFwd;
+                    }
+					else if(x == LENGTH-5) {
```
Comment (genericwoods) [inline src/main.c:878]: I mentioned this in our DM convo, but I think we can clean up some of the existing logic here (that’s made more confusing with each playback mode added) by using `if(x == LENGTH-mPingRep)` here (we _may_ need to explicitly assign numerical values to each of the enums, forgive me that I’m a bit rusty on vanilla C programming)

File: src/main.c (hunk 1 above)
Comment (GoneCaving) [reply src/main.c:864]: Really? Looks ok here.

**Comment by genericwoods** [comment]
To capture my response on the Lines forum, I think dropping the repeating mode makes the most sense (retains consistency with the larger grid, and keeps the code simpler)

File: src/main.c (hunk 2 above)
Comment (GoneCaving) [reply src/main.c:878]: The enums are 0 up, so should be easy enough. If we don't mind the overhead of setting the mPingFwd regardless of mode, we might be able to combine all of the modes into a single block (LENGTH-x).

File: src/main.c (hunk 1 above)
Comment (genericwoods) [reply src/main.c:864]: Just to make sure we’re seeing the same thing:
screen shot 2016-10-25 at 11 25 48 am

Also appears to be off on lines 310-343

File: src/main.c (hunk 2 above)
Comment (genericwoods) [reply src/main.c:878]: Even better, that seems like a good idea to me

**Comment by GoneCaving** [comment]
OK, abandoning this, and will adopt the comments above (and hopefully address the tabs vs spaces issue).

**Pull request closed without merging**